migrations/lib/importer/copy_step.rb

```diff
@@ -67,6 +67,7 @@ def initialize(intermediate_db, discourse_db, shared_data)
     def execute
       super
       with_progressbar(total_count) { copy_data }
+      after_copy_data
```

Comment:
I’m wondering if we should also add a `before_copy_data` hook. Right now, most pre-copy actions are handled by extending `execute`. If we’re introducing a dedicated hook for post-copy hook (which feels more intuitive), should we also add one for pre-copy actions to stay consistent?

The main reason I haven’t added it yet is, I'm concerned about the extra empty method dispatch overhead this would introduce in `CopyStep` for all steps. I know it's negligible compared to everything else but if we can avoid it why not?

Another consideration is naming. Should the hook be tightly tied to the `copy_data` method, or would a more generic name make sense?